src/SMatrix.jl

```diff
@@ -29,10 +29,6 @@ immutable SMatrix{S1, S2, T, L} <: StaticMatrix{T}
 end
 
 @generated function check_smatrix_params{S1,S2,L}(::Type{Val{S1}}, ::Type{Val{S2}}, T, ::Type{Val{L}})
-    if !(T <: DataType) # I think the way types are handled in generated fnctions might have changed in 0.5?
```

Comment:
Good point (I was treating it as a type parameter, but clearly it's not).

If you do want to have a look, feel free to target v0.6 only. We should also try change this to a `@pure` function, if possible (I had trouble on v0.5). Also if we can remove the `Val`s and split it into two functions - one to check `T` and another to check `S1,S2,L`, that would be fantastic.